### PR TITLE
osquery python module

### PR DESCRIPTION
**Pull request by marpaia**
Check out https://github.com/marpaia/osquery-python to see this using the normal GitHub UX. 

## Helping

If anyone would like to help pick off these tasks, feel free to submit a pull request to https://github.com/marpaia/osquery-python

## TODO

- [x] add a deregister API
- [x] write tests
- [x] see if `pip install thrift` installs the thrift compiler (*it doesn't*)
- [x] move thrift code inline so the `thrift` binary isn't required for building
- [x] add a table plugin example
- [x] add a config plugin example
- [x] add a logger plugin example
- [x] add a README
- [x] add routing for config plugins
- [x] add routing for logger plugins
- [x] improve routing for table plugins
- [x] add install instructions to readme
- [x] pep8 lint. maybe set `python setup.py lint` to run pylint.
- [x] change readme to rst and have setup.py include the readme

## Reach goals

- [ ] add a method to ConfigPlugin to allow async updates
- [ ] write integration tests

### CODE REVIEW

**Comment by arirubinstein** [comment]
rambo original

**Comment by theopolis** [comment]
So we are including the thrift-generated code inline? Does this allow for a 100% Python install? Meaning there are no system package manager dependencies or install-time compilation tasks?

**Comment by marpaia** [comment]
@theopolis correct, none at all. just the thrift and argparse python modules are required, but that is satisfied by the setup.py.

File: osquery/management.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+except ImportError:
+    print("Cannot import argparse: pip install argparse")
+    exit(1)
+
+try:
+    from thrift.transport import TSocket
+    from thrift.transport import TTransport
+    from thrift.protocol import TBinaryProtocol
+    from thrift.server import TServer
+except ImportError:
```
Comment (marpaia) [inline osquery/management.py]: Is this catch here necessary? Thoughts, @theopolis?

Comment (marpaia) [inline osquery/management.py]: argparse and thrift are installed via setup.py, so this should never be hit. also, not sure if an exit is more useful than an ImportError

File: osquery/management.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    parser.add_argument(
+        "--socket",
+        type=str,
+        default=DEFAULT_SOCKET_PATH,
+        help="Path to the extensions UNIX domain socket")
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=0,
+        help="XXX")
```
Comment (marpaia) [inline osquery/management.py]: Fix this

File: osquery/management.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    parser.add_argument(
+        "--interval",
+        type=int,
+        default=0,
+        help="XXX")
```
Comment (marpaia) [inline osquery/management.py]: Fix this

File: osquery/management.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    parser.add_argument(
+        "--verbose",
+        action="store_true",
+        help="XXX")
```
Comment (marpaia) [inline osquery/management.py]: Fix this

**Pull request merged**